src/main/java/ch/njol/skript/expressions/ExprDirection.java

```diff
@@ -124,89 +125,104 @@ public boolean init(final Expression<?>[] exprs, final int matchedPattern, final
 			case 4:
 				yaw = Math.PI / 2 * parseResult.mark;
 				horizontal = matchedPattern == 4;
+				break;
+			case 5:
+				from = (Expression<Location>) exprs[0];
+				relativeTo = exprs[1];
+				return true; // No amount
 		}
+		amount = (Expression<Number>) exprs[0];
 		return true;
 	}
-	
+
 	@Override
 	@Nullable
-	protected Direction[] get(final Event e) {
-		final Number n = amount != null ? amount.getSingle(e) : 1;
-		if (n == null)
+	protected Direction[] get(Event event) {
+		Number number = amount != null ? amount.getSingle(event) : 1;
+		if (number == null)
 			return new Direction[0];
-		final double ln = n.doubleValue();
+		double length = number.doubleValue();
 		if (direction != null) {
-			final Vector v = direction.clone().multiply(ln);
-			ExprDirection d = next;
-			while (d != null) {
-				final Number n2 = d.amount != null ? d.amount.getSingle(e) : 1;
-				if (n2 == null)
+			Vector vector = this.direction.clone().multiply(length);
+			ExprDirection addDirection = next;
+			while (addDirection != null) {
+				Number number2 = addDirection.amount != null ? addDirection.amount.getSingle(event) : 1;
```

Comment:
needs a better name than number 2
maybe something like delta length, or length modifier, or something like that.